src/MouseMoveProvider.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+import React, { Component, PropTypes } from 'react';
+import { uuid } from './utils';
+import invariant from 'invariant';
+
+type Subscription = { uuid: string, callback: () => void };
+type SubscriptionArray = [Subscription];
+
+export default class MouseMoveProvider extends Component {
+  subs: SubscriptionArray = [];
+  hasSubs: Boolean = false;
+  initialOffset: Number = window.pageYOffset;
```

Comment:
We can't access `window` here as it would break server rendering. This should be done in `componentDidMount`